src/api/app/models/bs_request.rb

```diff
@@ -107,6 +108,21 @@ def check_creator
     errors.add(:creator, "Login #{user.login} is not an active user")
   end
 
+  def double_actions
```

Comment:
What about introducing two new scopes

```
has_many :bs_request_actions, -> { includes([:bs_request_action_accept_info]) }, dependent: :destroy do
  def without_add_role
    where.not(type: 'add_role')
  end

  def without_maintenance_incidents
    where.not(type: 'maintenance_incident')
  end
end
```

and then we could use it in the validation:

```
bs_request_actions.each do |actionA|
   bs_request_actions.each do |actionB|
     next if actionA == actionB
     if actionA.type == actionB.type \
        && actionA.target_project == actionB.target_project \
        && actionA.target_package == actionB.target_package
       errors.add(:invalid, "double action for #{actionB.try(:target_project)}/#{actionB.try(:target_package)} ")
     end
   end
end
```